lib/galaxy/tools/actions/__init__.py

```diff
@@ -628,7 +628,7 @@ def set_metadata_defaults( self, output, dataset, tool, on_text, trans, incoming
         """
         if output.actions:
             for action in output.actions.actions:
-                if action.tag == "metadata":
+                if action.tag == "metadata" and action.default:
                     metadata_new_value = fill_template( action.default, context=params ).split(",")
```

Comment:
```
from galaxy.util.template import fill_template

fill_template(None, dict())
```

return the string `'<Cheetah.Template.Template object at 0x7f3bec07d890>'`, so I don't think there could be any use case.